powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/OperationServiceBehavior.java

```diff
@@ -575,10 +577,21 @@ private boolean isDataEqual(@NotNull OperationEntity operation, String providedD
     @Transactional
     public void expireOperations() {
         LockAssert.assertLocked();
-        final Date currentTimestamp = new Date();
         logger.debug("Running scheduled task for expiring operations");
-        try (final Stream<OperationEntity> pendingOperations = operationRepository.findExpiredPendingOperations(currentTimestamp)) {
-            pendingOperations.forEach(op -> expireOperation(op, currentTimestamp));
-        }
+
+        final Date currentTimestamp = new Date();
+        List<String> expiringOperationIds = operationRepository.findExpiredPendingOperationIds(currentTimestamp);
+        if (expiringOperationIds.isEmpty())
+            return;
+        int updatedRows = operationRepository.setExpiredToPendingOperations(expiringOperationIds);
+        if (expiringOperationIds.size() > updatedRows)
+            logger.warn("There was less updated rows ({}) than was expected ({}), ids on input: {}", expiringOperationIds.size(), updatedRows, expiringOperationIds);
+
+        List<OperationEntity> pendingOperations = operationRepository.findOperationsById(expiringOperationIds);
+        pendingOperations.stream().forEach(op-> {
+            if (EXPIRED.equals(op.getStatus()))
```

Comment:
Let's add an `{` and `}` block even for one line statements to avoid an unexpected mistake.